scripts/data/utreexo.py

```diff
@@ -0,0 +1,121 @@
+from poseidon_py.poseidon_hash import poseidon_hash_many
+
+# The array of trees in the forest
+# [T_1, T_2, T_4, T_8, ... ]
+root_nodes = [None] * 27
+
+# The set of leaf nodes in the forest
+leaf_nodes = dict()
```

Comment:
It is not convenient to use this module because of the global variables. Can you please refactor this into a class that encapsulates roots/leaves and provides methods for adding/removing items + generating proofs?